src/MongoDB.Driver/MongoDatabaseImpl.cs

```diff
@@ -260,10 +260,10 @@ public override TResult RunCommand<TResult>(IClientSessionHandle session, Comman
         {
             Ensure.IsNotNull(session, nameof(session));
             Ensure.IsNotNull(command, nameof(command));
-            readPreference = readPreference ?? ReadPreference.Primary;
 
             var operation = CreateRunCommandOperation(command);
-            return ExecuteReadOperation(session, operation, readPreference, cancellationToken);
+            var effectiveReadPreference = ReadPreferenceResolver.GetEffectiveReadPreference(session, readPreference, ReadPreference.Primary);
```

Comment:
Note that the explicitReadPreference is the readPreference parameter to the method, and the defaultReadPreference is Primary (not the ReadPreference from the MongoDatabaseSettings).